core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/dimension/legacy/IndexStorageEncoder.java

```diff
@@ -20,43 +20,86 @@
 import java.io.ByteArrayOutputStream;
 import java.io.DataOutputStream;
 import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
 
 import org.apache.carbondata.core.constants.CarbonCommonConstants;
-import org.apache.carbondata.core.datastore.columnar.BlockIndexerStorage;
+import org.apache.carbondata.core.datastore.columnar.PageIndexGenerator;
 import org.apache.carbondata.core.datastore.page.ColumnPage;
 import org.apache.carbondata.core.datastore.page.encoding.ColumnPageEncoder;
 import org.apache.carbondata.core.datastore.page.encoding.ColumnPageEncoderMeta;
+import org.apache.carbondata.core.datastore.page.encoding.EncodedColumnPage;
+import org.apache.carbondata.core.keygenerator.KeyGenerator;
 import org.apache.carbondata.core.memory.MemoryException;
+import org.apache.carbondata.core.metadata.datatype.DataType;
+import org.apache.carbondata.core.util.DataTypeUtil;
+import org.apache.carbondata.format.BlockletMinMaxIndex;
 import org.apache.carbondata.format.DataChunk2;
+import org.apache.carbondata.format.Encoding;
 import org.apache.carbondata.format.SortState;
 
 public abstract class IndexStorageEncoder extends ColumnPageEncoder {
-  BlockIndexerStorage indexStorage;
+  /**
+   * index generator
+   */
+  PageIndexGenerator pageIndexGenerator;
+  /**
+   * compressed data
+   */
   byte[] compressedDataPage;
 
-  abstract void encodeIndexStorage(ColumnPage inputPage);
+  /**
+   * encoded data page, in case of
+   */
+  EncodedColumnPage encodedColumnPage;
+
+  /**
+   * whether to store offset for column data
+   */
+  private boolean storeOffset;
```

Comment:
Please correct the name to appropriate. Bettter name as `isAdaptiveOnLength`

Reply:
ok